resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceTypeHelper.scala

```diff
@@ -0,0 +1,150 @@
+          "org.apache.hadoop.yarn.api.records.ResourceInformation")
+        val setResourceInformationMethod =
+          resource.getClass.getMethod("setResourceInformation", classOf[String],
+            resInfoClass)
+
+        val resourceInformation =
+          createResourceInformation(name, amount, unit, resInfoClass)
+        setResourceInformationMethod.invoke(resource, name, resourceInformation)
+      } catch {
+        case _: MatchError => throw new IllegalArgumentException(
+          s"Value of resource type should match pattern $AMOUNT_AND_UNIT_REGEX, " +
+              s"unmatched value: $rawAmount")
+        case e: InvocationTargetException =>
+          if (e.getCause != null) {
+            throw e.getCause
+          } else {
+            throw e
+          }
+      }
+    }
+    resource
+  }
+
+  def getCustomResourcesAsStrings(resource: Resource): String = {
+    if (resource == null) {
+      throw new IllegalArgumentException("Resource parameter should not be null!")
+    }
+
+    if (!ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+      return ""
+    }
+
+    var res: String = ""
+    try {
+      val resUtilsClass = Utils.classForName(
+        "org.apache.hadoop.yarn.util.resource.ResourceUtils")
+      val getNumberOfResourceTypesMethod = resUtilsClass.getMethod("getNumberOfKnownResourceTypes")
+      val numberOfResourceTypes: Int = getNumberOfResourceTypesMethod.invoke(null).asInstanceOf[Int]
+      val resourceClass = Utils.classForName(
+        "org.apache.hadoop.yarn.api.records.Resource")
+
+      // skip memory and vcores (index 0 and 1)
+      for (i <- 2 until numberOfResourceTypes) {
+        val getResourceInfoMethod = resourceClass.getMethod("getResourceInformation", JInteger.TYPE)
+        res ++= getResourceInfoMethod.invoke(resource, i.asInstanceOf[JInteger]).toString()
+        res ++= ", "
+      }
+    } catch {
+      case e: InvocationTargetException =>
+        if (e.getCause != null) {
+          throw e.getCause
+        } else {
+          throw e
+        }
+    }
+    res
+  }
+
+  private def createResourceInformation(
+      resourceName: String,
+      amount: Long,
+      unit: String,
+      resInfoClass: Class[_]) = {
+    val resourceInformation =
+      if (unit.nonEmpty) {
+        val resInfoNewInstanceMethod = resInfoClass.getMethod("newInstance",
+          classOf[String], classOf[String], JLong.TYPE)
+        resInfoNewInstanceMethod.invoke(null, resourceName, unit, amount.asInstanceOf[JLong])
+      } else {
+        val resInfoNewInstanceMethod = resInfoClass.getMethod("newInstance",
+          classOf[String], JLong.TYPE)
+        resInfoNewInstanceMethod.invoke(null, resourceName, amount.asInstanceOf[JLong])
+      }
+    resourceInformation
+  }
+
+  /**
+   * Checks whether Hadoop 2.x or 3 is used as a dependency.
+   * In case of Hadoop 3 and later,
+   * the ResourceInformation class should be available on the classpath.
```

Comment:
The line breaks in your comments are weird in many places. You're mixing really short with longer lines which makes the comments look weird.

Reply:
Could you please point to an example of an acceptable multiline commenting style with line breaks? 
Please check how I modified the comment.
Thanks